### PR TITLE
[RDE-71] feat(admin): export quiz statistics as CSV

### PR DESCRIPTION
**Pull request by Ki3mONo**
## RDE-71 - Quiz statistics CSV export

Closes RDE-71.

Lets an admin export a quiz's basic statistics as a CSV file from the result detail view.

### What changed

- **Frontend (`lib/results/csvExport.ts`)**: pure helpers to aggregate participant scores
  (`computeScoreStats` -> count / average / min / max), RFC 4180 CSV escaping, and a CSV
  builder that emits a UTF-8 BOM, a header row, and one summary row.
- **Frontend (`ResultDetailView`)**: an "Eksportuj CSV" button that downloads the file, plus
  a small stats summary (participants, average, min, max) above the scores table.

This PR is frontend-only; no backend changes are needed.

### Why client-side CSV

The CSV is generated in the browser from the already-fetched result archive. This keeps the
admin's local timezone for the date column (per the acceptance criteria) and avoids a second
authenticated, file-streaming endpoint.

### CSV format

One header row + one summary row, BOM-prefixed so spreadsheets read Polish characters:

```
Data quizu,Nazwa quizu,Liczba uczestników,Średni wynik,Minimalny wynik,Maksymalny wynik
2026-06-01 12:00:00,Przykładowy quiz,3,20,10,30
```

### Acceptance criteria

- [x] Admin sees an export option for a quiz's statistics.
- [x] Clicking generates a CSV file.
- [x] The file has correct headers and data.
- [x] Export also works for a quiz with no participants (count 0, empty stat cells).
- [x] Dates in the export use the admin's timezone (formatted client-side).

### Testing

- `frontend/lib/results/csvExport.test.ts`: aggregation (incl. empty quiz), CSV escaping,
  header/row output, BOM, empty-quiz row.
- `pnpm -C frontend validate` and `pnpm -C frontend test` pass (lint, prettier, type-check).

## Release Notes

* **New Features**
  * Results detail view now displays aggregated score statistics including participant count, average score, minimum, and maximum values.
  * Added CSV export functionality with a new button to download quiz statistics as a formatted file with Polish locale formatting.

* **Tests**
  * Added comprehensive test suite for CSV export utilities, including stats calculations, formatting, and edge cases.

### CODE REVIEW

File: backend/services/results.py [hunk 1]
```diff
@@ -102,7 +102,7 @@ def list_results_in_day(paths: StoragePaths, date: str) -> list[ResultFileMetada
                     score_count=len(data.get("scores", [])),
                 )
             )
-        except json.JSONDecodeError, OSError:
+        except (json.JSONDecodeError, OSError):
```
[the remark this answers is not in the document]
Comment (Ki3mONo) [reply backend/services/results.py]: Resolved by splitting into two separate `except` clauses (`except json.JSONDecodeError:` / `except OSError:`) rather than running `ruff format`. ruff 0.15.9's formatter rewrites `except (A, B):` into `except A, B:`, which is a SyntaxError on Python 3, so applying the formatter here would break the build. The two-clause form passes both `ruff format --check` and `ruff check`.

[the remark this answers is not in the document]
Comment (Ki3mONo) [reply backend/services/results.py]: Correction to my earlier reply: I was wrong. The project targets Python 3.14, where `except A, B:` without parentheses is valid PEP 758 syntax, so the original file was never broken and ruff's bare form is correct. My local check used Python 3.12 (anaconda), which gave a false SyntaxError. I've reverted `backend/services/results.py` to its original content - this PR no longer touches it, and CI is green.

**Review by OlaszPL — APPROVED** [review]
Tested locally, works fine. Code is clear too - approved! 🔥
